File: src/main/java/org/jdc/template/json/DateTimeStringDeserializer.java
package org.jdc.template.json;

import com.fasterxml.jackson.core.JsonParser;
import com.fasterxml.jackson.databind.DeserializationContext;
import com.fasterxml.jackson.databind.JsonDeserializer;

import org.threeten.bp.LocalDateTime;
import org.threeten.bp.format.DateTimeFormatter;

import java.io.IOException;

public class DateTimeStringDeserializer extends JsonDeserializer<LocalDateTime> {
    public static final DateTimeFormatter FORMATTER = DateTimeFormatter.ISO_DATE_TIME;

    @Override
    public LocalDateTime deserialize(JsonParser jsonParser, DeserializationContext ctxt) throws IOException {
        String dateString = jsonParser.getValueAsString(null);
        if (dateString != null) {
            return LocalDateTime.parse(jsonParser.getText(), FORMATTER);
        } else {
            return null;
        }
    }
}

File: src/main/kotlin/org/jdc/template/json/DateTimeStringDeserializer.kt
package org.jdc.template.json

import com.fasterxml.jackson.core.JsonParser
import com.fasterxml.jackson.databind.DeserializationContext
import com.fasterxml.jackson.databind.JsonDeserializer

import org.threeten.bp.LocalDateTime
import org.threeten.bp.format.DateTimeFormatter

import java.io.IOException

class DateTimeStringDeserializer : JsonDeserializer<LocalDateTime>() {

    @Throws(IOException::class)
    override fun deserialize(jsonParser: JsonParser, context: DeserializationContext): LocalDateTime? {
        val dateString = jsonParser.getValueAsString(null)
        return if (dateString != null) {
            LocalDateTime.parse(jsonParser.text, FORMATTER)
        } else {
            null
        }
    }

    companion object {
        val FORMATTER = DateTimeFormatter.ISO_DATE_TIME
    }
}
